src/allocators/allocator-guarantees.md

```diff
@@ -0,0 +1,58 @@
+## Recommendation for allocator implementations
+
+### Allocating memory
+
+In addition to implementing the conventational invariants ensuring the
+mutually exclusive allocation of memory, CHERI-aware implementations of
+`malloc()` and `calloc()` must return a capability that has the following
+properties:
+
+ * Is valid (i.e., with its tag bit set)
+ * Is unsealed
+ * Has bounds that permit access to the full requested range of the allocation
+ * Has bounds that do not permit access to any other current allocation,
+   implementing non-aliasing spatial safety
```

Comment:
"This includes any allocator metadata about the current allocation" or such, perhaps?

Reply:
Now added.

Reply:
I've added this but generalised in a couple of ways (1) On realloc() make sure to clear newly available memory and (2) Don't allow access to any allocator metadata, whether for the current allocation or otherwise. Check to make sure you are happy with the revised words, please!